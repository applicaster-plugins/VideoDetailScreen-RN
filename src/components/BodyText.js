import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { View, ViewPropTypes } from 'react-native';
import Text from '@applicaster/london-rn-components/src/SafeText';
import { labels, parameters } from '../utils';
import { styles } from './BodyText.styles';

class BodyText extends PureComponent {
  constructor(props) {
    super(props);
    // deferred for layout data to be ready
    this.styles = styles();
  }

  render() {
    const { style, children, collapseText, toggleCollapseText } = this.props;
    const { compactDescription } = parameters;
    return (
      <View style={[this.styles.container, style]}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={this.styles.text}
          numberOfLines={!compactDescription || collapseText ? undefined : 3}
        >
          {children}
        </Text>
        {compactDescription && (
          <Text style={this.styles.toggle} onPress={toggleCollapseText}>
            {collapseText ? labels.fewer : labels.more}
          </Text>
        )}
      </View>
    );
  }
}

BodyText.defaultProps = { style: {} };
BodyText.propTypes = {
  children: PropTypes.string,
  style: ViewPropTypes.style,
  toggleCollapseText: PropTypes.func,
  collapseText: PropTypes.bool
};

export default BodyText;
