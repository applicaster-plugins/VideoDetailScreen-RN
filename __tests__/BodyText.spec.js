import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import BodyText from '../src/components/BodyText';

jest.mock('../src/utils/zappConfig');

const style = {
  borderBottomWidth: 1,
  borderBottomColor: 'rgb(224,224,224)',
  paddingBottom: 20
};

it('BodyText renders correctly', () => {
  const toggleCollapseText = jest.fn();
  const tree = renderer
    .create(
      <BodyText
        style={style}
        collapseText={true}
        toggleCollapseText={toggleCollapseText}
      >
        Test text
      </BodyText>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
