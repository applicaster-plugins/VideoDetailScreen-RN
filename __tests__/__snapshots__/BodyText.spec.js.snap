// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyText renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "justifyContent": "flex-start",
        "marginTop": 12,
        "width": "99%",
      },
      Object {
        "borderBottomColor": "rgb(224,224,224)",
        "borderBottomWidth": 1,
        "paddingBottom": 20,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "rgb(135,136,147)",
        "fontFamily": "Exo2-Regular",
        "fontSize": 15,
        "fontStyle": "normal",
        "fontWeight": "normal",
        "letterSpacing": 0,
        "lineHeight": 22,
        "textAlign": "left",
      }
    }
  >
    Test text
  </Text>
</View>
`;
